Datum/Public/Get-FileProviderData.ps1

```diff
@@ -15,25 +15,70 @@ function Get-FileProviderData {
 
     process {
         $File = Get-Item -Path $Path
-        if($script:FileProviderDataCache.ContainsKey($File.FullName) -and 
+        if($script:FileProviderDataCache.ContainsKey($File.FullName) -and
         $File.LastWriteTime -eq $script:FileProviderDataCache[$File.FullName].Metadata.LastWriteTime) {
             Write-Verbose "Getting File Provider Cache for Path: $Path"
-            $script:FileProviderDataCache[$File.FullName].Value
+            Write-Output $script:FileProviderDataCache[$File.FullName].Value -NoEnumerate
         }
         else {
             Write-Verbose "Getting File Provider Data for Path: $Path"
-            $Data = switch ($File.Extension) {
-                '.psd1' { Import-PowerShellDataFile $File           | ConvertTo-Datum -DatumHandlers $DatumHandlers }
-                '.json' { ConvertFrom-Json (Get-Content -Raw $Path) | ConvertTo-Datum -DatumHandlers $DatumHandlers }
-                '.yml'  { ConvertFrom-Yaml (Get-Content -raw $Path) -ordered | ConvertTo-Datum -DatumHandlers $DatumHandlers }
-                
-                Default { Get-Content -Raw $Path }
+            $data = switch ($File.Extension) {
+                '.psd1' {
+                    $result = Import-PowerShellDataFile -Path $File | ConvertTo-Datum -DatumHandlers $DatumHandlers
+                    Write-Output $result -NoEnumerate
+                }
+                '.json' { 
+                    $result = ConvertFrom-Json (Get-Content -Path $Path -Raw) | ConvertTo-Datum -DatumHandlers $DatumHandlers
+                    Write-Output $result -NoEnumerate
+                }
+                '.yml'  { 
+                    $result = ConvertFrom-Yaml (Get-Content -Path $Path -Raw) -Ordered | ConvertTo-Datum -DatumHandlers $DatumHandlers
+                    Write-Output $result -NoEnumerate
+                }
+                '.csv'  {
```

Comment:
As we discussed previously, I don't want to add native support for CSV.
That said, I can imagine scenarios where it might be required (csv or other format), so I'd rather have an extensible system, similar to DatumHandlers (Module exposing functionalities defined in Datum.yml).

Also, Datum handlers _could_ be a way to support csv in some cases.